constraint/pkg/apis/templates/crd_schema.go

```diff
@@ -0,0 +1,60 @@
+package templates
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+var ConstraintTemplateSchemas map[string]*schema.Structural
+
+func init() {
+	ConstraintTemplateSchemas = make(map[string]*schema.Structural)
+
+	// Ingest the constraint template CRD for use in defaulting functions
+	crdJSON, err := yaml.YAMLToJSONStrict([]byte(constraintTemplateCRDYaml))
+	if err != nil {
+		panic(errors.Wrap(err, "Failed to convert Constraint Template yaml to JSON"))
+	}
+
+	unCRD := unstructured.Unstructured{}
+	unCRD.UnmarshalJSON(crdJSON)
+	constraintTemplateCRD := &apiextensionsv1.CustomResourceDefinition{}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(unCRD.Object, constraintTemplateCRD)
+	if err != nil {
+		panic(errors.Wrap(err, "Failed to convert unstructured CRD to apiextensions.CustomResourceDefinition{}"))
+	}
+
+	// NewStructural requires apiextensions.JSONSchemaProps, where ConstraintTemplate uses
+	// apiextensionsv1.JSONSchemaProps.  Set up scheme for conversion.
+	scheme := runtime.NewScheme()
+	if err := apiextensionsv1.AddToScheme(scheme); err != nil {
+		panic(err)
+	}
+	if err := apiextensions.AddToScheme(scheme); err != nil {
+		panic(err)
+	}
+
+	// Fill version map with Structural types derived from ConstraintTemplate versions
+	for _, crdVersion := range constraintTemplateCRD.Spec.Versions {
+		versionedSchemaCopy := crdVersion.Schema.OpenAPIV3Schema.DeepCopy()
```

Comment:
Do we need to deepcopy this? It seems like we're throwing it away after-the-fact anyway?

Reply:
Removed.